src/app/modules/item/components/user-progress/user-progress.component.ts

```diff
@@ -7,9 +8,9 @@ import { OverlayPanel } from 'primeng/overlaypanel';
   templateUrl: './user-progress.component.html',
   styleUrls: [ './user-progress.component.scss' ]
 })
-export class UserProgressComponent implements OnChanges {
+export class UserProgressComponent implements OnInit, OnChanges {
 
-  @Input() userProgress?: TeamUserProgress;
+  @Input() userProgress!: TeamUserProgress;
```

Comment:
Does it really help in some way? Have you measured an impact? 
If not, I don't really like that this kind of type assertion which are not guaranteed at compile time.

Reply:
When I measured it (on my computer) for the worst case scenario, it represented ~2 seconds. Initial load was 24s (still on my computer)
I could use [this technique](https://stackoverflow.com/questions/35528395/make-directive-input-required/54240378#54240378)

Reply:
What helps is removing the `*ngIf="userProgress"` in the template

Reply:
I didn't use the `comp[inp]` technique on purpose until now but now I am re-reading it, it seems quite good actually.

Reply:
Fixed